src/test/d4n_filter_test.py

```diff
@@ -0,0 +1,234 @@
+import boto3
+import redis
+import unittest
+import subprocess
+import json
+
+# Assume a redis server and RGW are running
+
+access_key = 'test3'
+secret_key = 'test3'
+host = 'localhost'
+port = 8000
+
+endpoint_url = "http://%s:%d" % (host, port)
+
+client = boto3.client(service_name='s3',
+                aws_access_key_id=access_key,
+                aws_secret_access_key=secret_key,
+                endpoint_url=endpoint_url,
+                use_ssl=False,
+                verify=False)
+
+s3 = boto3.resource('s3', 
+                use_ssl=False,
+                verify=False,
+                endpoint_url=endpoint_url, 
+                aws_access_key_id=access_key,
+                aws_secret_access_key=secret_key)
+
+bucket = s3.Bucket('bkt')
+bucket_location = bucket.create()
+obj = s3.Object(bucket_name='bkt', key='test.txt')
+
+class D4NFilterTestCase(unittest.TestCase):
+    
+    '''
+    D4N Directory Unit Tests
+    '''
+
+    # Successful setValue Call and Redis Check
+    def test_set_value(self):
+        r = redis.Redis(host='localhost', port=6379, db=0, decode_responses=True)
```

Comment:
What happens in these tests when redis is not up? Can we check for failure here and then return a clear error message that the tests can't execute because redis isn't up?